BootloaderCorePkg/Stage2/Stage2Hob.c

```diff
@@ -605,6 +605,142 @@ BuildUniversalPayloadHob (
 }
 
 
+/**
+  Build SMM variable related HOBs for Universal Payload
+
+**/
+VOID
+BuildSmmVariableHobs (
+  VOID
+)
+{
+  EFI_STATUS                           Status;
+  LDR_SMM_INFO                         *SmmInfoHob;
+  UINT32                               Length;
+  PLD_SMM_REGISTERS                    *SmmRegisterHob;
+  PLD_GENERIC_REGISTER                 *Reg;
+  EFI_SMRAM_HOB_DESCRIPTOR_BLOCK       *SmmMemoryHob;
+  SPI_FLASH_INFO                       *SpiFlashInfoHob;
+  LOADER_PLATFORM_INFO                 *PlatformInfo;
+  NV_VARIABLE_INFO                     *NvVariableHob;
+  UINT32                               Base;
+
+  // Build SMM register HOB for universal payload
+  Length = sizeof (PLD_SMM_REGISTERS) + 5 * sizeof (PLD_GENERIC_REGISTER);
+  SmmRegisterHob = BuildGuidHob (&gPldSmmRegisterInfoGuid, Length);
+  if (SmmRegisterHob != NULL) {
+    ZeroMem (SmmRegisterHob, Length);
+    SmmRegisterHob->Revision = 0;
+    SmmRegisterHob->Count    = 5;
+
+    // Get SMM information HOB
+    SmmInfoHob = (LDR_SMM_INFO *)GetGuidHobData (NULL, NULL, &gSmmInformationGuid);
+    ASSERT (SmmInfoHob != NULL);

```

Comment:
Can we avoid ASSERT ?   I would prefer error flow handling instead.
If we cannot find the HOB, then skip build the HOB.